MFVNeutralino/test/histos.py

```diff
@@ -19,7 +24,68 @@
+process.mfvAnalysisCutsMETCutTightVtx = process.mfvAnalysisCuts.clone(min_nvertex = 1)
+process.mfvVertexHistosMETCutTightVtx = process.mfvVertexHistos.clone()
+process.pEventMETCutTightVtx = cms.Path(common * process.mfvAnalysisCutsMETCutTightVtx * process.mfvEventHistosMETCutTightVtx)
+process.pMETCutTightVtx = cms.Path(common * process.mfvAnalysisCutsMETCutTightVtx * process.mfvVertexHistosMETCutTightVtx)
+
+#B events with at least 1 Tight SV and not pass MET trigger and cut
+process.mfvEventHistosNoMETCutTightVtx = process.mfvEventHistos.clone()
+process.mfvAnalysisCutsNoMETCutTightVtx = process.mfvAnalysisCuts.clone(min_nvertex = 1, apply_presel = 6)
+process.mfvVertexHistosNoMETCutTightVtx = process.mfvVertexHistos.clone()
+process.pEventNoMETCutTightVtx = cms.Path(common * process.mfvAnalysisCutsNoMETCutTightVtx * process.mfvEventHistosNoMETCutTightVtx)
+process.pNoMETCutTightVtx = cms.Path(common * process.mfvAnalysisCutsNoMETCutTightVtx * process.mfvVertexHistosNoMETCutTightVtx)
+
+#C events with at least 1 Loose SV and pass MET trigger and cut
+process.mfvEventHistosMETCutLooseVtx = process.mfvEventHistos.clone()
+process.mfvAnalysisCutsMETCutLooseVtxTightCheck = process.mfvAnalysisCuts.clone(vertex_src = 'mfvSelectedVerticesTight', min_nvertex = 0, max_nvertex = 0) # make sure no tight Vtx survived
+process.mfvAnalysisCutsMETCutLooseVtx = process.mfvAnalysisCuts.clone(vertex_src = 'mfvSelectedVerticesTightNtk3or4', min_nvertex = 1, max_nvertex = 100000)
+process.mfvVertexHistosMETCutLooseVtx = process.mfvVertexHistos.clone(vertex_src = 'mfvSelectedVerticesTightNtk3or4')
+process.pEventMETCutLooseVtx = cms.Path(common * process.mfvAnalysisCutsMETCutLooseVtxTightCheck * process.mfvAnalysisCutsMETCutLooseVtx * process.mfvEventHistosMETCutLooseVtx)
+process.pMETCutLooseVtx = cms.Path(common * process.mfvAnalysisCutsMETCutLooseVtxTightCheck * process.mfvAnalysisCutsMETCutLooseVtx * process.mfvVertexHistosMETCutLooseVtx)
+
+#D events with at least 1 Loose SV and not pass MET trigger and cut
+process.mfvEventHistosNoMETCutLooseVtx = process.mfvEventHistos.clone()
+process.mfvAnalysisCutsNoMETCutLooseVtxTightCheck = process.mfvAnalysisCuts.clone(vertex_src = 'mfvSelectedVerticesTight', min_nvertex = 0, max_nvertex = 0, apply_presel = 6) # make sure no tight Vtx survived
+process.mfvAnalysisCutsNoMETCutLooseVtx = process.mfvAnalysisCuts.clone(vertex_src = 'mfvSelectedVerticesTightNtk3or4', min_nvertex = 1, max_nvertex = 100000, apply_presel = 6)
+process.mfvVertexHistosNoMETCutLooseVtx = process.mfvVertexHistos.clone(vertex_src = 'mfvSelectedVerticesTightNtk3or4')
+process.pEventNoMETCutLooseVtx = cms.Path(common * process.mfvAnalysisCutsNoMETCutLooseVtxTightCheck * process.mfvAnalysisCutsNoMETCutLooseVtx * process.mfvEventHistosNoMETCutLooseVtx)
+process.pNoMETCutLooseVtx = cms.Path(common * process.mfvAnalysisCutsNoMETCutLooseVtxTightCheck * process.mfvAnalysisCutsNoMETCutLooseVtx * process.mfvVertexHistosNoMETCutLooseVtx)
```

Comment:
Probably we should put all of these into an `if` statement pertaining only to the MET trigger (or maybe better yet, into a separate file where we import all of these?), along with the special input files you're using. The other n-1 plots are probably okay (we should make a note of what "ExtraLoose" is though).

Reply:
Actually most of those are out of date for me... I used to make those in histos for counting number of events but I am not doing that now for ML... So maybe I should just remove those in the future.